src/app/qgshandlebadlayers.cpp

```diff
@@ -509,3 +480,143 @@ int QgsHandleBadLayers::layerCount()
 {
   return mLayerList->rowCount();
 }
+
+QString QgsHandleBadLayers::findFile( const QString &fileName, const QString &basePath, int maxDepth, int driveMargin )
+{
+  int depth = 0;
+  QDir folder = QDir( basePath );
+  QString existingBase;
+  while ( !folder.exists() && folder.absolutePath().count( folder.separator() ) > driveMargin )
```

Comment:
folder.absolutePath().count( folder.separator() ) is fragile -- use QDir/QFileInfo classes to iterate up the directory chain instead.

Reply:
How would you suggest to validate the depth of the current folder as I couldn't find an alternative way to check the depth of the data. This is important as I don't want to get close to the root to perform search, this would stall the system and possibly provide wrong results.

Reply:
I think in this case it might be worth iterating up from the original path until there's no parents directories left (or a suitable cutoff is hit... e.g. 20 folders), and building this list in its entirety. Then you could trim off the last n entries from the list, where n == the minimum depth you want to allow from a root folder.

Reply:
Wouldn't that mean looping twice on things? Also QDir cannot cdUp if the directory does not exist, this would mean that I have to fiddle as I do when attempting to find the first existing directory. I'm not sure how optimal this is compared to checking for system separators. If you tell me that this is the only way, then I'm ok with it. It just might be a hassle to build depth index, keep the related path in order to properly set a stop when the path is encountered. :/

Reply:
I'll await your confirmation if you truly want me to loop over thing instead of using the / separator that is used in QT. I feel like the looper that you proposed would provide an useless overhead (in term of code, and runtime) when relying on the unix separator is robust.

Reply:
The tricky thing here is.... we can't gaurantee that projects use the correct slash for their current operating system :scream: There's potential for projects to be shared between Windows and MacOS, for instance. Even on a single operating system environment (linux), I still see windows style paths creep into layer sources in some cases.

It makes things super tricky! I wasn't aware that QDir couldn't iterate up through non-existing paths - that's unfortunate.

I'm not sure what the best approach forward is here. Maybe we should treat either \ or / the same on any platform, and consider them both as candidates for a new folder. What do you think? @m-kuhn any additional thoughts?

Reply:
> The tricky thing here is.... we can't gaurantee that projects use the correct slash for their current operating system scream There's potential for projects to be shared between Windows and MacOS, for instance. Even on a single operating system environment (linux), I still see windows style paths creep into layer sources in some cases.

Is this true? Wouldn't this lead to issue trying to open a Windows project under *nix systems?

> I'm not sure what the best approach forward is here. Maybe we should treat either \ or / the same on any platform, and consider them both as candidates for a new folder.

... but if it is, the approach of treating `\` and `/` equally everywhere sounds like the best bet to me.

Reply:
From what I gather (I might be wrong) if we rely on QDir/QFileInfo to save and parse filepath, this would ensure that every path is in the same format (in this instance the unix format), otherwise I would also assume that QDir and QFileInfo would do the proper conversion when receiving a path that is formatted for another system. ( Could anyone do a quick check on linux with a windows formatted path to validate this please?)

Since all paths would be converted to the same formatting, I would then assume that it is appropriate to relying on the unix separator in order to obtain folder depth.